pkg/config/config.go

```diff
@@ -223,6 +223,19 @@ func (c *Config) GetStepConfig(flagValues map[string]interface{}, paramJSON stri
 		stepConfig.mixIn(flagValues, filters.Parameters)
 	}
 
+	// fetch secrets from vault
+	vaultClient, err := getVaultClientFromConfig(stepConfig)
+	if err != nil {
+		return StepConfig{}, err
+	}
+	vaultConfig, err := getVaultConfig(vaultClient, stepConfig, parameters)
+	if err != nil {
+		return StepConfig{}, err
+	}
+	stepConfig.mixIn(vaultConfig, filters.General)
```

Comment:
we don't need to use `stepConfig.mixIn()` here. We get content for valid parameters only thus no filtering required.

I guess it is sufficient to loop over `vaultConfig` and add to `stepConfig`.
Alternatively one could think of passing `stepConfig` as a pointer into `getVaultConfig()` and alter the configuration directly there.
Maybe the function then needs a slight rename e.g. `addVaultConfig()`

What do you think?

Reply:
> I guess it is sufficient to loop over vaultConfig and add content to stepConfig.

 yes you are totally right.

> Alternatively one could think of passing stepConfig as a pointer into getVaultConfig() and alter the configuration directly there. 

Technically it wouldn't be necessary to make a `stepConfig` pointer since the the `Config` property is a map which are always passed as references. But I guess in this way it's more transparent that the map inside the `stepParameter` gets modified.

>  Maybe the function then needs a slight rename e.g. addVaultCredentials()

I agree